ts/src/index.ts

```diff
@@ -129,13 +155,21 @@ export async function start(config: Config = {}): Promise<void> {
         config);
     return;
   }
+
   let normalizedConfig: ProfilerConfig;
   try {
-    normalizedConfig = await initConfig(config);
+    normalizedConfig = await initConfigAndMaybeStartHeapProfiler(config);
   } catch (e) {
     logError(`Could not start profiler: ${e}`, config);
     return;
   }
+
+  // Stop heap profiler if, after initialization, the config indicates that
+  // the heap profiler is disabled.
+  if (normalizedConfig.disableHeap) {
```

Comment:
I don't think this case is possible any more, or is it? `disableHeap` can be synchronously configured only. `initConfigMetadata` cannot change the value of `disableHeap`.

Either remove, or change this to an assertion?

Reply:
Done.

Reply:
Done.